ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -95,22 +117,80 @@ module Json = struct
       | _ ->
           concated
     in
-    List.map
-      (fun field ->
-        let ty = string_of_ty field.ty in
-        `O
-          [
-            ("name", `String (concat_and_convert field))
-          ; ("description", `String (String.trim field.field_description))
-          ; ("type", `String ty)
-          ]
+    List.fold_left
+      (fun (fields, enums) field ->
+        let ty, e = string_of_ty_with_enums field.ty in
+        ( `O
+            [
+              ("name", `String (concat_and_convert field))
+            ; ("description", `String (String.trim field.field_description))
+            ; ("type", `String ty)
+            ]
+          :: fields
+        , enums @ e
+        )
       )
-      fields
+      ([], []) fields
+
+  let enums_from_result obj msg =
+    match msg.msg_result with
+    | None ->
+        []
+    | Some (t, _d) ->
+        if obj.name = "event" && String.lowercase_ascii msg.msg_name = "from"
+        then
+          []
+        else
+          let _, enums = string_of_ty_with_enums t in
+          enums
+
+  let enums_from_params ps =
+    List.fold_left
+      (fun enums p ->
+        let _t, e = string_of_ty_with_enums p.param_type in
+        enums @ e
+      )
+      [] ps
+
+  let session_id =
+    {
+      param_type= Ref Datamodel_common._session
+    ; param_name= "session_id"
+    ; param_doc= "Reference to a valid session"
+    ; param_release= Datamodel_common.rio_release
+    ; param_default= None
+    }
+
+  let enums_in_messages_of_obj obj =
+    List.fold_left
+      (fun enums msg ->
+        let params =
+          if msg.msg_session then
+            session_id :: msg.msg_params
+          else
+            msg.msg_params
+        in
+        let enums1 = enums_from_result obj msg in
+        let enums2 = enums_from_params params in
+        enums @ enums1 @ enums2
```

Comment:
While it's not important here, `List.concat [list1, list2, ...]` is preferred because it's more efficient